client/hijack.go

```diff
@@ -72,11 +74,23 @@ func (cli *Client) postHijacked(ctx context.Context, path string, query url.Valu
 	defer clientconn.Close()
 
 	// Server hijacks the connection, error 'connection closed' expected
-	_, err = clientconn.Do(req)
-
-	rwc, br := clientconn.Hijack()
+	resp, err := clientconn.Do(req)
+	if resp != nil {
```

Comment:
I think the previous comment https://github.com/moby/moby/pull/33262#discussion_r117484500 meant to say that we don't have to check `resp` if `err != nil`; we can still return early here, to reduce complexity, e.g.;

```go
resp, err := clientconn.Do(req)
if err != nil {
	return types.HijackedResponse{}, err
}

defer resp.Body.Close()

switch resp.StatusCode {
case http.StatusOK, http.StatusSwitchingProtocols:
	rwc, br := clientconn.Hijack()
	return types.HijackedResponse{Conn: rwc, Reader: br}, err
}
body, err := ioutil.ReadAll(resp.Body)
if err != nil {
	return types.HijackedResponse{}, err
}
return types.HijackedResponse{}, fmt.Errorf("Error response from daemon: %s", bytes.TrimSpace(body))
```